### ISSUE TITLE
macOS SecureTransport support breaks in 7.87.0

### ISSUE THREAD

**Issue by percontation** [curl/curl#10348]
Curl version 7.87.0 appears to break `./configure --with-secure-transport` support on my machine.

### I did this

On macOS Monterey 12.6.3, using the GitHub release and `./configure --with-secure-transport && make -j`:

curl 7.87.0 breaks with any of a variety of error messages:
```
curl-7.87.0$ ./src/curl -V
curl 7.87.0 (x86_64-apple-darwin21.6.0) libcurl/7.87.0 SecureTransport zlib/1.2.13 brotli/1.0.9 zstd/1.5.2 libidn2/2.3.4 libpsl/0.21.2 (+libidn2/2.3.4) nghttp2/1.51.0
Release-Date: 2022-12-21
Protocols: dict file ftp ftps gopher gophers http https imap imaps ldap ldaps mqtt pop3 pop3s rtsp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS brotli HSTS HTTP2 HTTPS-proxy IDN IPv6 Largefile libz NTLM NTLM_WB PSL SSL threadsafe UnixSockets zstd
curl-7.87.0$ ./src/curl -Ss https://github.com > /dev/null
curl: (35) Internal SSL engine error encountered during the SSL handshake
curl-7.87.0$ ./src/curl -Ss https://github.com > /dev/null
curl: (56) SSLRead() return error -9806
curl-7.87.0$ ./src/curl -Ss https://github.com > /dev/null
curl: (56) SSLRead() return error -50
curl-7.87.0$ ./src/curl -Ss https://github.com > /dev/null
curl: (56) SSLRead() return error -50
curl-7.87.0$ ./src/curl -Ss https://github.com > /dev/null
curl: (56) SSLRead() return error -9806
curl-7.87.0$ ./src/curl -Ss https://github.com > /dev/null
curl: (56) SSLRead() return error -9806
curl-7.87.0$ ./src/curl -Ss https://github.com > /dev/null
curl: (56) SSLRead() return error -50
```

while curl 7.86.0 works fine:
```
curl-7.86.0$ ./src/curl -V
curl 7.86.0 (x86_64-apple-darwin21.6.0) libcurl/7.86.0 SecureTransport zlib/1.2.13 brotli/1.0.9 zstd/1.5.2 libidn2/2.3.4 libpsl/0.21.2 (+libidn2/2.3.4) nghttp2/1.51.0
Release-Date: 2022-10-26
Protocols: dict file ftp ftps gopher gophers http https imap imaps ldap ldaps mqtt pop3 pop3s rtsp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS brotli HSTS HTTP2 IDN IPv6 Largefile libz NTLM NTLM_WB PSL SSL threadsafe UnixSockets zstd
curl-7.86.0$ ./src/curl -Ss https://github.com > /dev/null
curl-7.86.0$ ./src/curl -Ss https://github.com > /dev/null
curl-7.86.0$ ./src/curl -Ss https://github.com > /dev/null
```

TLS servers behave differently. Some seem to always fail with some error (like github.com), but for other servers, TLS requests work like 50% of the time with curl 7.87.0.

**Comment by percontation** [comment]
FWIW: The issue reproduces in 55807e6c0, but its parent commit dbd74baf7 seems unaffected. Not surprising, since 55807e6c0 was the most recent change to `lib/vtls/sectransp.c`.

**Comment by jay** [comment]
paramErr (-50) and errSSLClosedAbort (-9806). Probably a duplicate of #10227. Please try master branch.

/cc @icing

**Comment by percontation** [comment]
yes, confirmed, it's a dupe, already fixed in 16bb32e10 as per that issue.

(I thought I tested master before complaining but apparently not 🤦. Sorry!)
